Modules/Module_Processing/Module_ASL/xASL_wrp_ProcessM0.m

```diff
@@ -120,10 +131,40 @@ function xASL_wrp_ProcessM0(x)
 
 			% Find the smallest of the previously selected positive nonzero TEs
 			minTE = min(uniqueNonzeroTE);
-			fprintf('Multi-TE M0 present. Using the shortest TE=%.1f ms for M0 calibration.\n', minTE*1000);
+			% JSON values were converted from BIDS (seconds) to Legacy (millisecond) during JSON reading
+			fprintf('Multi-TE M0 present. Using the shortest TE = %.1f ms for M0 calibration.\n', minTE);
+
+			% Find the correct indices and update the jsonM0 parameter
+			indMinTE = jsonM0.Q.EchoTime == minTE;
+			jsonM0.Q.EchoTime = jsonM0.Q.EchoTime(indMinTE);
+			if isfield(jsonM0, 'RepetitionTime') && length(jsonM0.RepetitionTime) > 1
+				jsonM0.RepetitionTime = jsonM0.RepetitionTime(indMinTE);
+			end
+
+			% Select the minimal TEs
+			imM0 = imM0(:, :, :, indMinTE);
+		end
+	end
 
-			% Average M0 image across all these minimal TEs
-			imM0 = xASL_stat_MeanNan(imM0(:,:,:,jsonM0.Q.EchoTime == minTE), 4);
+%% 0D. Use the M0 with the TR closest to 2000 ms (if we have multiple M0s)
+	if isfield(jsonM0, 'RepetitionTime')
+		uniqueNonzeroTR = unique(jsonM0.RepetitionTime(jsonM0.RepetitionTime > 0));

```

Comment:
What do we do when we have additional M0s with zeroes? Perhaps it never happens, but now you added the '> 0' clause specifically, we probably need a specific warning as well if there are M0s with zeroes?
Or do we in case of a single M0 with a zero just ignore it.

Reply:
Sorry. that's a historical thing. sometimes we have TE vector with some zeros in it. They don't correspond to any volume and are in DICOM only and are supposed to be removed. But this happens already in the import. Plus PLD of M0 in BIDS is supposed to be 0. So when looking for a PLD vector, we skipped 0s. 

So I was just copy pasting code. Here, there's no need to have this check anymore. I'm removing it for clarity.